doc/source/tune/tutorials/tune_get_data_in_and_out.md

```diff
@@ -0,0 +1,685 @@
+
+Ray AIR (which contains Ray Tune) provides a [`Checkpoint`](air-checkpoints-doc) API for that purpose. `Checkpoint` objects can be created from various sources (dictionaries, directories, cloud storage) and used between different AIR components.
+
+In Tune, `Checkpoints` are created by the user in their Trainable functions and reported using the optional `checkpoint` argument of `session.report`. `Checkpoints` can contain arbitrary data and can be freely passed around the Ray cluster. After a tuning run is over, `Checkpoints` can be [obtained from the results](/tune/examples/tune_analyze_results).
+
+Ray Tune can be configured to [automatically sync checkpoints to cloud storage](tune-checkpoint-syncing), {class}`keep only a certain number of checkpoints to save space <~ray.air.config.CheckpointConfig>` and more.
+
+```{note}
+The experiment state itself is checkpointed separately. See {ref}`tune-two-types-of-ckpt` for more details.
+```
+
+In our example, we want to be able to resume the training from the latest checkpoint, and to save the `trained_model` in a checkpoint every iteration. To accomplish this, we will use the `session` and `Checkpoint` APIs.
+
+
+```python
+from ray.air import Checkpoint
+
+
+def training_function(config, data):
+    model = {
+        "hyperparameter_a": config["hyperparameter_a"],
+        "hyperparameter_b": config["hyperparameter_b"],
+    }
+    epochs = config["epochs"]
+
+    # Load the checkpoint, if there is any.
+    loaded_checkpoint = session.get_checkpoint()
+    if loaded_checkpoint is not None:
+        last_epoch = loaded_checkpoint["epoch"] + 1
+    else:
+        last_epoch = 0
+
+    # Simulate training & evaluation - we obtain back a "metric" and a "trained_model".
+    for epoch in range(last_epoch, epochs):
+        # Simulate doing something expensive.
+        time.sleep(1)
+        metric = (0.1 + model["hyperparameter_a"] * epoch / 100) ** (
+            -1
+        ) + model["hyperparameter_b"] * 0.1 * data["A"].sum()
+        trained_model = {"state": model, "epoch": epoch}
+
+        # Create the checkpoint.
+        checkpoint = Checkpoint.from_dict({"model": trained_model})
+        session.report(metrics={"metric": metric}, checkpoint=checkpoint)
+
+
+tuner = Tuner(
+    tune.with_parameters(training_function, data=data),
+    param_space={
+        "hyperparameter_a": tune.uniform(0, 20),
+        "hyperparameter_b": tune.uniform(-100, 100),
+        "epochs": 10,
+    },
+    tune_config=tune.TuneConfig(num_samples=4, metric="metric", mode="max"),
+    run_config=RunConfig(
+        callbacks=[MLflowLoggerCallback(experiment_name="example")]
+    ),
+)
+```
+
+With all of those changes implemented, we can now run our tuning and obtain meaningful metrics and artifacts.
+
+
+```python
+results = tuner.fit()
+results.get_dataframe()
+```
+
+
+<div class="tuneStatus">
```

Comment:
Can we remove this tuner output and only show the dataframe? The table shows similar information and is not formatted that well in the markdown.